File: app/android/app/src/main/java/com/barghest/bux/domain/model/Analytics.kt
package com.barghest.bux.domain.model

import java.math.BigDecimal

data class TransactionSummary(
    val totalIncome: BigDecimal,
    val totalExpense: BigDecimal,
    val net: BigDecimal,
    val byCategory: List<CategorySummary>,
    val byMonth: List<MonthlySummary>
)

data class CategorySummary(
    val categoryId: Int?,
    val categoryName: String,
    val categoryIcon: String,
    val categoryColor: String,
    val type: String,
    val total: BigDecimal,
    val count: Int
)

data class MonthlySummary(
    val year: Int,
    val month: Int,
    val income: BigDecimal,
    val expense: BigDecimal,
    val net: BigDecimal
)

data class AssetGroup(
    val type: AccountType,
    val label: String,
    val totalBalance: BigDecimal,
    val currency: String,
    val accounts: List<Account>
)

data class NetWorthData(
    val totalByCurrency: Map<String, BigDecimal>,
    val assetGroups: List<AssetGroup>,
    val investmentValue: BigDecimal,
    val investmentCurrency: String
)

File: server/services/transaction/internal/presentation/http/dto/analytics.go
package dto

type CategorySummary struct {
	CategoryID    *uint  `json:"category_id"`
	CategoryName  string `json:"category_name"`
	CategoryIcon  string `json:"category_icon"`
	CategoryColor string `json:"category_color"`
	Type          string `json:"type"`
	Total         string `json:"total"`
	Count         int64  `json:"count"`
}

type MonthlySummary struct {
	Year    int    `json:"year"`
	Month   int    `json:"month"`
	Income  string `json:"income"`
	Expense string `json:"expense"`
	Net     string `json:"net"`
}

type TransactionSummaryResponse struct {
	TotalIncome  string            `json:"total_income"`
	TotalExpense string            `json:"total_expense"`
	Net          string            `json:"net"`
	ByCategory   []CategorySummary `json:"by_category"`
	ByMonth      []MonthlySummary  `json:"by_month"`
}
